### PR TITLE
[SPARK-27051] [Core] Bump Jackson version to 2.9.8

### PR DESCRIPTION
**Pull request by yanboliang**
## What changes were proposed in this pull request?
Fasterxml Jackson version before 2.9.8 is affected by multiple [CVEs](https://github.com/FasterXML/jackson-databind/issues/2186), we need to fix bump the dependent Jackson to 2.9.8.

## How was this patch tested?
Existing tests and offline benchmark. 
I have run ```SPARK_GENERATE_BENCHMARK_FILES=1 build/sbt "sql/test:runMain org.apache.spark.sql.execution.datasources.json.JSONBenchmark"``` to check there is no performance degradation for this upgrade.

### CODE REVIEW

File: core/pom.xml
Comment (gatorsmile) [inline core/pom.xml]: ?

Comment (yanboliang) [reply core/pom.xml]: Jackson 2.9.8 is not dependent on scala-reflect (but 2.9.6 have this dependency), but Spark core needs this dependency, so we should add it explicitly.

**Comment by gatorsmile** [comment]
LGTM pending Jenkins

File: core/pom.xml
Comment (gatorsmile) [reply core/pom.xml]: I confirmed it.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM, too.

**Comment by dongjoon-hyun** [comment]
cc @dbtsai since he is a release manager and this is related to CVEs.

**Comment by dbtsai** [comment]
Do we need to cut another rc for 2.4.1 because of this?

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**
